### PR TITLE
DTSERWFOUR-211 - Fetch transfer method information and display for ed…

### PR DESCRIPTION
**Pull request by rajkumar-balakrishnasamy**

### CODE REVIEW

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/HyperwalletTransferMethodUi.java [hunk 1]
```diff
@@ -131,6 +133,26 @@ public Intent getIntentAddTransferMethodActivity(@NonNull final Context context,
...
+     * @param lockScreenToPortrait if set <code>true</code> screen will be locked to Portrait mode;
+     *                             otherwise <code>false</code> screen will follow whatever the
+     *                             device orientation is directed.
+     * @return an Intent with the data necessary to launch the {@link UpdateTransferMethodActivity}
+     */
+    public Intent getIntentUpdateTransferMethodActivity(@NonNull final Context context,
+            @NonNull final String transferMethodType, @NonNull final String transferMethodToken,
+            final boolean lockScreenToPortrait) {
+        Intent intent = new Intent(context, UpdateTransferMethodActivity.class);
+        intent.putExtra(EXTRA_TRANSFER_METHOD_TYPE, transferMethodType);
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/HyperwalletTransferMethodUi.java:150]: This is wrong, integrator will only pass transferMethodToken, as I explained in planning transferMethodType you will need to get from graphql response

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/ListTransferMethodFragment.java [hunk 2]
```diff
@@ -373,6 +373,20 @@ void bind(@NonNull final TransferMethod transferMethod) {
...
                 }
 
+                //  Todo: Remove before code merge. Used for Debugging purpose.
+                itemView.setOnClickListener(new OneClickListener() {
+                    @Override
+                    public void onOneClick(View v) {
+                        String type = transferMethod.getField(TransferMethod.TransferMethodFields.TYPE);
+                        String token = transferMethod.getField(TransferMethod.TransferMethodFields.TOKEN);
+                        Intent intent = new Intent(itemView.getContext(), UpdateTransferMethodActivity.class);
+                        intent.putExtra(UpdateTransferMethodActivity.EXTRA_TRANSFER_METHOD_TYPE, type);
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/ListTransferMethodFragment.java:383]: incorrect, see above comment

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/UpdateTransferMethodActivity.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        setContentView(R.layout.activity_update_transfer_method);
+
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE);
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+        getSupportActionBar().setTitle(TransferMethodUtils.getTransferMethodName(this,
+                getIntent().getStringExtra(EXTRA_TRANSFER_METHOD_TYPE)));
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/UpdateTransferMethodActivity.java:55]: get it from graphql response

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java [hunk 4]
```diff
@@ -69,7 +69,7 @@ String formatToApi(@NonNull final String displayValue) {
      * @return a String formatted to the specification in the mask pattern
      */
     String formatToDisplay(@NonNull final String apiValue) {
-        if (mField != null && mField.getMask() != null) {
+        if (mField != null && mField.getMask() != null && !mField.isFieldValueMasked()) {
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java:72]: This logic is only for update flow. Make sure you are not breaking other flows

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/UpdateTransferMethodActivity.java (hunk 3 above)
Comment (rajkumar-balakrishnasamy) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/UpdateTransferMethodActivity.java:55]: Done.

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java (hunk 4 above)
Comment (rajkumar-balakrishnasamy) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java:72]: Unable to move this condition specific to update. Will request the QA to validate the add scenario.

**Pull request merged**